src/main/java/com/zaxxer/nuprocess/osx/ProcessKqueue.java

```diff
@@ -33,13 +36,12 @@
  */
 final class ProcessKqueue extends BaseEventProcessor<OsxProcess>
 {
-   private static final int KEVENT_POOL_SIZE = 16;
+   private static final int NUM_KEVENTS = 64;
    private static final TimeSpec timeSpec;
    private static final int JAVA_PID;
 
    private volatile int kqueue;
-   private Kevent triggeredEvent;
-   private BlockingQueue<Kevent> keventPool;
+   private ThreadLocal<KeventArray> keventArrays;
```

Comment:
This not an issue if the only interaction is this ThreadLocal is by a NuProcess managed thread (see comments on ``registerProcess(OsxProcess process)`` below).

However, the [``BaseEventProcess.run()``](https://github.com/brettwooldridge/NuProcess/blob/master/src/main/java/com/zaxxer/nuprocess/internal/BaseEventProcessor.java#L61) method *must* then have a ``finally`` that calls a shutdown/cleanup method so that:

 * ``KeventArray.free()`` gets called, and
 * ``keventArrays.remove()`` is called to remove the ThreadLocal